Fresh install not working on brand new ember app

For a fresh ember app, is it expected you need to `npm install --save-dev typescript` before `ember install ember-cli-typescript@latest` ?

When installing on a brand new app I get an error, 

**Error: Cannot find module 'typescript'**

Looks like the README mentions that `typescript` will be installed by the ember install command.

- https://github.com/typed-ember/ember-cli-typescript#setup-and-configuration

I tried `ember install ember-cli-typescript@latest` on Mac, Windows, and Ubuntu.

**Steps to reproduce:**

1. Use `nvm` or Node.js installer to install the current LTS version of Node
2. `npm install -g ember-cli`
3. `ember new name-your-app`
4. `cd name-your-app`
5. `ember install ember-cli-typescript@latest`

**Expected Result**

No errors,

```
⚡ember install ember-cli-typescript@latest
npm: Installed ember-cli-typescript@latest
installing ember-cli-typescript
  create app/config/environment.d.ts
  create tsconfig.json
  create types/file-watching/index.d.ts
  create types/ember-data.d.ts
  install packages typescript, @types/ember, @types/rsvp, @types/ember-test-helpers, @types/ember-testing-helpers, @types/ember-data, @types/ember-qunit, @types/qunit
npm: Installed typescript@latest, @types/ember@latest and 6 other packages
Installed addon package.
```

**Actual Result**

```
⚡ ember install ember-cli-typescript@latest
Cannot find module 'typescript'


Stack Trace and Error Report: /var/folders/fy/h2t9mc1d15d57zh1vtt5f1h00000gn/T/error.dump.421e772c9c853d6e626c69a9571649c0.log
```

**Attempt to resolve:**

Simple to fix but perhaps this needs to be added to the documentation:

1. `npm install --save-dev typescript`
2. `ember install ember-cli-typescript@latest`

_Below is a full log of my attempt on a mac..._

```
billheaton ~/Code/pixelhandler
⚡ nvm install --lts
Installing latest LTS version.
Downloading and installing node v8.11.1...
Downloading https://nodejs.org/dist/v8.11.1/node-v8.11.1-darwin-x64.tar.xz...
######################################################################## 100.0%
Computing checksum with shasum -a 256
Checksums matched!
Now using node v8.11.1 (npm v5.6.0)
billheaton ~/Code/pixelhandler
⚡ node -v
v8.11.1
billheaton ~/Code/pixelhandler
⚡ npm install -g ember-cli
/Users/billheat/.nvm/versions/node/v8.11.1/bin/ember -> /Users/billheat/.nvm/versions/node/v8.11.1/lib/node_modules/ember-cli/bin/ember

> fsevents@1.2.3 install /Users/billheat/.nvm/versions/node/v8.11.1/lib/node_modules/ember-cli/node_modules/fsevents
> node install

[fsevents] Success: "/Users/billheat/.nvm/versions/node/v8.11.1/lib/node_modules/ember-cli/node_modules/fsevents/lib/binding/Release/node-v57-darwin-x64/fse.node" already installed
Pass --update-binary to reinstall or --build-from-source to recompile

> spawn-sync@1.0.15 postinstall /Users/billheat/.nvm/versions/node/v8.11.1/lib/node_modules/ember-cli/node_modules/spawn-sync
> node postinstall

+ ember-cli@3.1.4
added 803 packages in 25.638s
billheaton ~/Code/pixelhandler
⚡ ember new file-watching
installing app
  create .editorconfig
  create .ember-cli
  create .eslintignore
  create .eslintrc.js
  create .travis.yml
  create .watchmanconfig
  create README.md
  create app/app.js
  create app/components/.gitkeep
  create app/controllers/.gitkeep
  create app/helpers/.gitkeep
  create app/index.html
  create app/models/.gitkeep
  create app/resolver.js
  create app/router.js
  create app/routes/.gitkeep
  create app/styles/app.css
  create app/templates/application.hbs
  create app/templates/components/.gitkeep
  create config/environment.js
  create config/targets.js
  create ember-cli-build.js
  create .gitignore
  create package.json
  create public/robots.txt
  create testem.js
  create tests/helpers/.gitkeep
  create tests/index.html
  create tests/integration/.gitkeep
  create tests/test-helper.js
  create tests/unit/.gitkeep
  create vendor/.gitkeep
npm: Installed dependencies
Successfully initialized git.
billheaton ~/Code/pixelhandler
⚡ cd file-watching/
billheaton ~/Code/pixelhandler/file-watching ±master
⚡ ls
README.md		ember-cli-build.js	package.json		tests
app			node_modules		public			vendor
config			package-lock.json	testem.js
billheaton ~/Code/pixelhandler/file-watching ±master
⚡ ember install ember-cli-typescript@latest
npm: Installed ember-cli-typescript@latest
Cannot find module 'typescript'


Stack Trace and Error Report: /var/folders/fy/h2t9mc1d15d57zh1vtt5f1h00000gn/T/error.dump.ff0896a0c48dfbc5ea7d46aebcb45438.log
billheaton ~/Code/pixelhandler/file-watching ±master
```

**Error report:**

```
⚡ cat /var/folders/fy/h2t9mc1d15d57zh1vtt5f1h00000gn/T/error.dump.ff0896a0c48dfbc5ea7d46aebcb45438.log
=================================================================================

ENV Summary:

  TIME: Fri May 04 2018 16:08:59 GMT-0700 (PDT)
  TITLE: ember
  ARGV:
  - /Users/billheat/.nvm/versions/node/v8.11.1/bin/node
  - /Users/billheat/.nvm/versions/node/v8.11.1/bin/ember
  - install
  - ember-cli-typescript@latest
  EXEC_PATH: /Users/billheat/.nvm/versions/node/v8.11.1/bin/node
  TMPDIR: /var/folders/fy/h2t9mc1d15d57zh1vtt5f1h00000gn/T
  SHELL: /bin/bash
  PATH:
  - /Users/billheat/.nvm/versions/node/v8.11.1/bin
  - /usr/local/bin
  - /usr/bin
  - /bin
  - /usr/sbin
  - /sbin
  - /usr/local/bin
  - /usr/local/sbin
  - /usr/local/bin
  - /usr/bin
  - /bin
  - /usr/sbin
  - /sbin
  PLATFORM: darwin x64
  FREEMEM: 128172032
  TOTALMEM: 17179869184
  UPTIME: 60062
  LOADAVG: 30.94775390625,100.63720703125,61.2373046875
  CPUS:
  - Intel(R) Core(TM) i7-5557U CPU @ 3.10GHz - 3100
  - Intel(R) Core(TM) i7-5557U CPU @ 3.10GHz - 3100
  - Intel(R) Core(TM) i7-5557U CPU @ 3.10GHz - 3100
  - Intel(R) Core(TM) i7-5557U CPU @ 3.10GHz - 3100
  ENDIANNESS: LE
  VERSIONS:
  - ares: 1.10.1-DEV
  - cldr: 32.0
  - http_parser: 2.8.0
  - icu: 60.1
  - modules: 57
  - nghttp2: 1.25.0
  - node: 8.11.1
  - openssl: 1.0.2o
  - tz: 2017c
  - unicode: 10.0
  - uv: 1.19.1
  - v8: 6.2.414.50
  - zlib: 1.2.11

ERROR Summary:

  - broccoliBuilderErrorStack: [undefined]
  - codeFrame: [undefined]
  - errorMessage: Cannot find module 'typescript'
  - errorType: [undefined]
  - location:
    - column: [undefined]
    - file: [undefined]
    - line: [undefined]
  - message: Cannot find module 'typescript'
  - name: Error
  - nodeAnnotation: [undefined]
  - nodeName: [undefined]
  - originalErrorMessage: [undefined]
  - stack: Error: Cannot find module 'typescript'
    at Function.Module._resolveFilename (module.js:547:15)
    at Function.Module._load (module.js:474:25)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/billheat/Code/pixelhandler/file-watching/node_modules/ember-cli-typescript/lib/incremental-typescript-compiler/index.js:10:12)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)

=================================================================================
```

--- a/lib/incremental-typescript-compiler/index.js
+++ b/lib/incremental-typescript-compiler/index.js
@@ -7,7 +7,6 @@ const path = require('path');
 const fs = require('fs-extra');
 const resolve = require('resolve');
 const compile = require('../utilities/compile');
-const ts = require('typescript');
 const TypescriptOutput = require('./typescript-output-plugin');
 const CompilerState = require('./compiler-state');
 
@@ -28,6 +27,7 @@ module.exports = class IncrementalTypescriptCompiler {
     this.addons = this._discoverAddons(project, []);
     this.state = new CompilerState();
 
+    this._ts = project.require('typescript');
     this._watchProgram = null;
   }
 
@@ -125,10 +125,10 @@ module.exports = class IncrementalTypescriptCompiler {
   }
 
   _formatDiagnosticMessage(diagnostic) {
-    return ts.formatDiagnostic(diagnostic, {
+    return this._ts.formatDiagnostic(diagnostic, {
       getCanonicalFileName: path => path,
-      getCurrentDirectory: ts.sys.getCurrentDirectory,
-      getNewLine: () => ts.sys.newLine,
+      getCurrentDirectory: this._ts.sys.getCurrentDirectory,
+      getNewLine: () => this._ts.sys.newLine,
     });
   }
 

--- a/package.json
+++ b/package.json
@@ -96,9 +96,6 @@
   "resolutions": {
     "@types/ember": "2.8.13"
   },
-  "peerDependencies": {
-    "typescript": "^2.4.2"
-  },
   "engines": {
     "node": "^4.5 || 6.* || >= 7.*"
   },

